webf/lib/src/rendering/flex.dart

```diff
@@ -889,9 +893,51 @@ class RenderFlexLayout extends RenderLayoutBox {
       if (flexShrink > 0) {
         totalFlexShrink += flexShrink;
       }
+      if (isHorizontal && child is RenderBoxModel) {
+        if (child.renderStyle.overflowX == CSSOverflowType.hidden) {
+          overflowHiddenNodeId.add(childNodeId);
+          overflowHiddenNodeTotalMinWidth += child.contentConstraints?.minWidth ?? 0;
+        } else {
+          overflowNotHiddenNodeTotalWidth += _originalMainSize;
+        }
+      }
     }
 
     if (runChildren.isNotEmpty) {
+      if (overflowHiddenNodeId.isNotEmpty && runMainAxisExtent > flexLineLimit && overflowHiddenNodeTotalMinWidth < flexLineLimit) {
+        int _overflowNodeSize = overflowHiddenNodeId.length;
+        double overWidth = flexLineLimit - overflowNotHiddenNodeTotalWidth;
+        double avgOverWidth = overWidth / _overflowNodeSize;
+        int index = 0;
+        for (int nodeId in overflowHiddenNodeId) {
+          _RunChild? _runChild = runChildren[nodeId];
+          if (_runChild != null) {
+            index += 1;
+            if (overWidth <= 0) {// 剩余宽度少于0情况
+              BoxConstraints oldConstraints = _runChild.child.constraints;
+              double _minWidth = oldConstraints.minWidth;
+              _runChild.child.layout(BoxConstraints(minWidth: _minWidth, maxWidth: _minWidth, minHeight: oldConstraints.minHeight, maxHeight: oldConstraints.maxHeight), parentUsesSize: true);
+              _runChild.originalMainSize = _minWidth;
+              _childrenIntrinsicMainSizes[nodeId] = _minWidth;
+            } else if (_runChild.originalMainSize <= avgOverWidth) {// 子控件实际宽度小于剩余宽度的平均值情况
```

Comment:
// the actual width of a subwidget is less than the average value of the remaining width.